### PR TITLE
Added Keycloak+postgres

### PR DESCRIPTION
**Pull request by paoloantinori**
There is still a bizarre bug with Kustomize:

- file with postgres definitions fails if we use kustomize, but it succeed with an directly `oc apply -f postgres-keycloak.yaml`

It might even be a bug with Kustomize for what I can understand.

Note:

this is likely a brittle implementation since I had to explicitly put the hostname in the route configuration for keycloak, something that was required to have support for https, a hard requirement that keycloak has to access the admin UI. Likely this will need to be improved, but we have a Keycloak on Operate First!

(Sorry, I know you'd have preferred having someoneelse other than me working on this, but there should be enough bugs for you to have fun!)

### CODE REVIEW

File: manifests/registry/base/postgres-keycloak.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      labels:
+        statefulset: postgres-keycloak
+    spec:
+      containers:
+      - name: postgres
+        # image: postgres:11.1
+        image: quay.io/centos7/postgresql-12-centos7:1
+        env:
+        - name: POSTGRESQL_ADMIN_PASSWORD
+          value: admin1234
```
Comment (carlesarnal) [inline manifests/registry/base/postgres-keycloak.yaml:62]: maybe extract this to a secret like you're doing with the regular PostgreSQL password?

**Review by carlesarnal — APPROVED** [review]
LGTM

**Comment by paoloantinori** [comment]
I'm fine with the suggestion but I'm not planning to touch this soon, so if
you are happy to contribute on this now, you can give it a go, otherwise
let's file this as a future task.

More in general: we are using passwords in a public repo, something that is
obviously bad, but I was planning to change them later on.

**Comment by famartinrh** [comment]
this deploys a new postgresql database instead of using the existing one and creating a specific db for it, but anyway I'm ok as it is and I don't think that's a problem.

However, the keycloak deployment is included in the `base` kustomization, and I think we don't want that. Keycloak deployment is going to be used for the multitenancy deployment, so I think this should be instead in the `overlay_multitenancy` folder

**Comment by paoloantinori** [comment]
> this deploys a new postgresql database instead of using the existing one and creating a specific db for it, but anyway I'm ok as it is and I don't think that's a problem.

this was deliberate to keep it more portable. ideally a whole kc deployments is managed externally, so we might not even need to deploy our own one. 


> However, the keycloak deployment is included in the base kustomization, and I think we don't want that. Keycloak deployment is going to be used for the multitenancy deployment, so I think this should be instead in the overlay_multitenancy folder

I'd say yes and no. But I take your point. Ideally I wanted to isolate it entirely on its own module, but I've failed to do so, and I suspect I might have hit a `kustomize` bug. The same exact manifests fail if I deploy them with `oc apply -k` while they succeed if I just do a `oc apply -f`

**Review by EricWittmann — APPROVED** [review]

**Pull request merged**